flink-runtime/src/test/java/org/apache/flink/runtime/util/config/memory/ProcessMemoryUtilsTestBase.java

```diff
@@ -350,4 +368,35 @@ public MemorySize getTotalProcessMemorySize() {
 			throw new UnsupportedOperationException();
 		}
 	}
+
+	private static class DummyFlinkMemoryUtils implements FlinkMemoryUtils<DummyFlinkMemory> {
```

Comment:
Meta comment: `FlinkMemoryUtils` does not seem to be documented well enough. I would suggest that every interface method should get a JavaDoc explaining what it does. Also the JavaDocs for the implementations of `FlinkMemoryUtils` are scarce at most.

Reply:
True, `FlinkMemory` JavaDocs can include its structure and relationships between components. I am not sure whether the methods need any explanation afterwards as they are just getters. The same thing applies to `FlinkMemoryUtils`. Its JavaDocs can be improved. I will add a hot fix commit.